internal/test/daemon/daemon.go

```diff
@@ -282,37 +282,51 @@ func (d *Daemon) StartWithLogFile(out *os.File, providedArgs ...string) error {
 	defer ticker.Stop()
 	tick := ticker.C
 
+	timeout := time.NewTimer(5 * time.Second) // timeout for the whole loop
+	defer timeout.Stop()
+
+	iterTimeout := time.NewTimer(2 * time.Second) // timeout for each iteration
```

Comment:
we may want to extract the duration in a const or var :angel: (mainly because of the `.Reset` below)

Reply:
Even better, don't need this timeout at all.